mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp

```diff
@@ -614,9 +614,14 @@ static Value createLinalgBodyCalculationForElementwiseOp(
                                             args.front(), zero);
     }
 
-    if (isa<IntegerType>(srcTy) && isa<IntegerType>(dstTy) && bitExtend)
+    if (isa<IntegerType>(srcTy) && isa<IntegerType>(dstTy) && bitExtend) {
+      auto srcIntTy = dyn_cast<IntegerType>(srcTy);
+      if (srcIntTy.isUnsigned())
```

Comment:
```suggestion
      if (cast<IntegerType>(srcTy).isUnsigned())
```
Can also be two lines, but I think since you already know it is an IntegerType, `cast` is the better choice.

Reply:
I don't need the `srcIntTy` afterwards, so using `cast` is fine :)